state/protocol/badger/state.go

```diff
@@ -989,6 +989,48 @@ func IsBootstrapped(db storage.DB) (bool, error) {
 	return true, nil
 }
 
+// GetChainIDFromLatestFinalizedHeader attempts to retrieve the consensus chainID
+// from the latest finalized header in the database, before storage or protocol state have been initialized.
+// Expected errors during normal operations:
+// - [storage.ErrNotFound] if the node is not bootstrapped.
+func GetChainIDFromLatestFinalizedHeader(db storage.DB) (flow.ChainID, error) {
```

Comment:
Would suggest to rename this to `GetChainID`. How precisely we do that is largely in implementation detail, which is already covered in the documentation. This detail is not important enough in my opinion to emphasize it as part of the function name.